File: app/src/main/java/com/example/test/data/database/BinDatabase.java
package com.example.test.data.database;

import android.content.Context;

import com.example.test.data.model.CardEntity;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {CardEntity.class}, version = 6, exportSchema = false)
public abstract class BinDatabase extends RoomDatabase {

    private static final String DB_NAME = "card2.db";
    private static BinDatabase instance = null;

    public static BinDatabase getInstance(Context context) {
        if (instance == null) {
            instance = Room.databaseBuilder(context, BinDatabase.class, DB_NAME).build();
        }
        return instance;
    }
    public abstract BinDao binDao();
}

File: app/src/main/java/com/example/test/data/database/BinDatabase.kt
package com.example.test.data.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.example.test.data.model.CardEntity

@Database(entities = [CardEntity::class], version = 6, exportSchema = false)
abstract class BinDatabase : RoomDatabase() {

    companion object {

        private const val DB_NAME = "card2.db"

        @Volatile
        private var instance: BinDatabase? = null

        fun getInstance(context: Context): BinDatabase
        {
            if (instance == null) {
                instance = Room.databaseBuilder(
                    context.applicationContext,
                    BinDatabase::class.java,
                    DB_NAME).fallbackToDestructiveMigration().build()
            }
            return instance as BinDatabase
        }
    }

    abstract fun binDao(): BinDao
}